pyxem/utils/calibration_utils.py

```diff
@@ -0,0 +1,122 @@
+    
+    Parameters
+    ----------
+    patterns : hyperspy:Signal2D object
+        Diffration patterns to be iteratively matched to find maximum correlation scores.  Should be of known phase.
+    calibration_guess : float
+        Inital value for the diffraction calibration in inverse Angstoms per pixel.
+    library_phases : diffsims:StructureLibrary Object
+        Dictionary of structures and associated orientations for which
+        electron diffraction is to be simulated.  Used to create the DiffractionLibrary.
+    lib_gen : diffsims:DiffractionLibraryGenerator Object
+        Computes a library of electron diffraction patterns for specified atomic
+        structures and orientations.  Used to create the DiffractionLibrary.
+    size : integer
+        How many different steps to test for the first two iterations.
+    max_excitation_error : float
+        Gets passed to get_diffraction_library
+    **kwargs to be passed to index_dataset_with_template_rotation
+    
+    Returns
+    -------
+    mean_cal : float
+        Mean of calibrations found for each pattern.
+    full_corrlines : np.array of shape (size*2 + 20, 2 , number of patterns)
+        Gives the explicit correlation vs calibration values.
+    found_cals : np.array of shape (number of patterns)
+        List of optimal calibration values for each pattern.
+    """
+    
+    images = patterns
+    
+    num_patterns = images.data.shape[0]
+    found_cals = np.zeros((num_patterns))
+    found_cals[:] = calibration_guess
+    full_corrlines = np.zeros((0,2,num_patterns))
+    
+    
+    
+    stepsize = 0.01*calibration_guess
+    #first set of checks
+    corrlines = _calibration_iteration(images,calibration_guess,library_phases,lib_gen,stepsize,size,num_patterns,max_excitation_error,**kwargs)
+    full_corrlines = np.append(full_corrlines,corrlines,axis = 0)
+    
+    #refined calibration checks
+    calibration_guess = full_corrlines[full_corrlines[:,1,:].argmax(axis = 0),0,0].mean()
+    corrlines = _calibration_iteration(images,calibration_guess,library_phases,lib_gen,stepsize,size,num_patterns,max_excitation_error,**kwargs)
+    full_corrlines = np.append(full_corrlines,corrlines,axis = 0)
+
+    #more refined calibration checks with smaller step
+    stepsize = 0.001*calibration_guess
+    size = 20
+    calibration_guess = full_corrlines[full_corrlines[:,1,:].argmax(axis = 0),0,0].mean()    
+    
+    corrlines = _calibration_iteration(images,calibration_guess,library_phases,lib_gen,stepsize,size,num_patterns,max_excitation_error,**kwargs)
+    full_corrlines = np.append(full_corrlines,corrlines,axis = 0)
+    found_cals = full_corrlines[full_corrlines[:,1,:].argmax(axis = 0),0,0]
+    
+
+    
+    mean_cal = found_cals.mean()
+    return mean_cal,full_corrlines, found_cals
+
+def _calibration_iteration(images,calibration_guess,library_phases,lib_gen,stepsize,size,num_patterns,max_excitation_error,**kwargs):
```

Comment:
Would be nice to have a docstring explaining what this function does even though it is private. You can refer to the above function for the parameter explanations.